libs/core/execution_base/include/hpx/execution_base/completion_signatures.hpp

```diff
@@ -600,6 +606,58 @@ namespace hpx::execution::experimental {
     template <typename Sender, typename Env = empty_env>
     inline constexpr bool is_sender_v = is_sender<Sender, Env>::value;
 
+    namespace detail {
+
+        // Sender may or may not support sends_stopped
+        // In the false case only the value_types and error_types are checked
+        template <class S, class R, class CS,
+            bool sends_stopped = CS::sends_stopped>
+        struct sender_support_sends_stopped;
+
+        template <class S, class R, class CS>
+        struct sender_support_sends_stopped<S, R, CS, true>
+          : std::integral_constant<bool,
+                is_receiver_of_v<R, completion_signatures_of_t<S, env_of_t<R>>>>
+        {
+        };
+
+        template <class S, class R, class CS>
+        struct sender_support_sends_stopped<S, R, CS, false>
+          : std::integral_constant<bool,
+                is_invocable_variant_of_tuples<set_value_t, R,
+                    typename CS::template value_types<hpx::tuple,
```

Comment:
What if the sender's completion signatures don't expose a dependent type `value_types`?

Reply:
It will not reach this point as `is_sender_v` will evaluate as false. Thus `is_sender_of` is false